### PR TITLE
S-LSTM implementation for chainerx

### PR DESCRIPTION
**Pull request by dido1998**

### CODE REVIEW

**Comment by dido1998** [comment]
Can this be reviewed?

File: chainerx_cc/chainerx/python/array.cc [hunk 1]
```diff
@@ -51,13 +51,22 @@ namespace python {
...
 using internal::MoveArrayBody;
 
 }  // namespace
 
 namespace py = pybind11;
 using py::literals::operator""_a;
 
+py::tuple ToTuple(const std::vector<Array>& ary) {
+    py::tuple ret{ary.size()};
+    for (uint i = 0; i < ary.size(); i++) {
```
Comment (niboshi) [inline chainerx_cc/chainerx/python/array.cc]: Let's use `size_t` instead of `uint`.

**Comment by niboshi** [comment]
Please resolve conflict.

**Comment by dido1998** [comment]
The test failed for `Maximum` which is not related to this PR.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Among the test failures in Jenkins, this one seems relevant:
```
12:28:16 /repo/chainerx_cc/chainerx/routines/activation.cc:30:39: error: the parameter 'x' is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param,-warnings-as-errors]
12:28:16 std::vector<Array> ExtractGates(Array x, int64_t n_splits, int64_t axis) {
12:28:16                                 ~~~~~ ^
12:28:16                                 const &
```

**Comment by dido1998** [comment]
I have made the change

**Comment by niboshi** [comment]
Thanks
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Please fix doctest failure in Jenkins.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM
Thanks!

**Pull request merged**